File: src/main/java/mediathek/gui/actions/ShowDuplicateStatisticsAction.java
package mediathek.gui.actions;

import mediathek.gui.duplicates.statistics.DuplicateStatisticsDialog;
import org.jspecify.annotations.NonNull;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;

public class ShowDuplicateStatisticsAction extends AbstractAction {
    private final Frame owner;

    public ShowDuplicateStatisticsAction(@NonNull Frame owner) {
        this.owner = owner;
        putValue(Action.NAME, "Film-Statistik anzeigen");
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        var dlg = new DuplicateStatisticsDialog(owner, this);
        dlg.setVisible(true);
    }
}

File: src/main/kotlin/mediathek/gui/actions/ShowDuplicateStatisticsAction.kt
package mediathek.gui.actions

import mediathek.filmlisten.FilmCatalog
import mediathek.gui.duplicates.statistics.DuplicateStatisticsDialog
import java.awt.Frame
import java.awt.event.ActionEvent
import javax.swing.AbstractAction

class ShowDuplicateStatisticsAction(
    private val owner: Frame,
    private val filmCatalog: FilmCatalog,
) : AbstractAction() {
    init {
        putValue(NAME, "Film-Statistik anzeigen")
    }

    override fun actionPerformed(event: ActionEvent?) {
        DuplicateStatisticsDialog(owner, filmCatalog, this).isVisible = true
    }
}
